pkg/vm/stackitem/item.go

```diff
@@ -795,6 +797,10 @@ func (i *Map) Drop(index int) {
 func IsValidMapKey(key Item) bool {
 	switch key.(type) {
 	case *Bool, *BigInteger, *ByteArray:
+		bytes, err := key.TryBytes()
```

Comment:
I think it can optimized a bit. Booleans are limited to two values and integers can only be 256-bit wide, both types will always fit into the limit (hence we don't need to convert them to bytes), the only problematic case is `ByteArray`.